### PR TITLE
Feature/authenticationlogout

### PR DESCRIPTION
**Pull request by jackmott**
Adds deauthenticate and deauthenticateWithLogin functions to mirror the existing authenticate functions. Also adds to the authenticate testcase to ensure that protected content can not be accessed after deauthenticate.

### CODE REVIEW

File: src/Suave/Authentication.fs [hunk 1]
```diff
@@ -98,8 +99,15 @@ let authenticated relativeExpiry secure : WebPart =
...
                  succeed)
 
-//  let deauthenticate : WebPart =
-//    Cookies.unset_cookies
+let deauthenticate _ : WebPart =
+  unsetPair SessionAuthCookie 
+  >=> unsetPair StateCookie
+
+let deauthenticateWithLogin loginPage : WebPart =
+  deauthenticate()
```
Comment (haf) [inline src/Suave/Authentication.fs]: I think you can drop the `()` and the `_` from `deauthenticate` as a WebPart is a function to start with.

File: src/Suave/Authentication.fsi [hunk 2]
```diff
@@ -41,6 +41,13 @@ val authenticateWithLogin : relativeExpiry:CookieLife
                           -> fSuccess:WebPart
                           -> WebPart
 
+val deauthenticate : _
```
Comment (haf) [inline src/Suave/Authentication.fsi]: Docs?

File: src/Suave/Authentication.fsi [hunk 3]
```diff
@@ -41,6 +41,13 @@ val authenticateWithLogin : relativeExpiry:CookieLife
...
+                   -> WebPart
+
+val deauthenticateWithLogin : loginPage :string
```
Comment (haf) [inline src/Suave/Authentication.fsi]: Docs?

**Comment by haf** [comment]
Looking sweet! Warm welcome to the Suave universe!

**Comment by jackmott** [comment]
I'm going to close this, fix the issues, and resubmit a PR on the master branch instead. Thanks.

**Pull request closed without merging**